cli/lsp/code_lens.rs

```diff
@@ -222,7 +221,7 @@ async fn resolve_implementation_code_lens(
 ) -> Result<lsp::CodeLens, AnyError> {
   let asset_or_doc =
     language_server.get_cached_asset_or_document(&data.specifier)?;
-  let line_index = asset_or_doc.line_index();
+  let line_index = (*asset_or_doc.line_index()).clone();
```

Comment:
The `line_index` should be immutable in the `Arc`, so why do we need to clone the contents inside the `Arc`?

Reply:
oh yeah, you got a point there... hmmmm... it must be related to situations where we dynamically get the line index _after_ the result